### ISSUE TITLE
Incorrect reflow fix for list item with MkDocs-flavored admonition

### ISSUE THREAD

**Issue by sisp** [rvben/rumdl#469]
### rumdl version

0.1.29

### Installation method

Pre-built binary

### Bug description

Reformatting a Markdown document with a list item containing an MkDocs-flavored admonition leads to (a) incorrect Markdown formatting of the admonition and (b) ignores wrapping long lines in the admonition's body.

### Steps to reproduce

1. Create a file `test.md` with this content:

    ```markdown
    # List item with MkDocs admonition

    - Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.

        !!! note

            Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    ```

1. Run `rumdl check --fix test.md`

1. See the reformatted file content.

### Expected behavior

The reformatting change should be this:

```diff
 # List item with MkDocs admonition

-- Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
+- Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
+  incididunt ut labore et dolore magna aliqua.

     !!! note

-        Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
+        Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi
+        ut aliquip ex ea commodo consequat.
```

### Actual behavior

The current reformatting change is this:

```diff
 # List item with MkDocs admonition

-- Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
+- Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
+  incididunt ut labore et dolore magna aliqua.

-    !!! note
+  !!! note

         Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
```

There are two problems:

1. The admonition header `!!! note` is dedented by 2 spaces.
1. The admonition body is not line-wrapped.

The reformatted content is incorrect MkDocs-flavored Markdown:

```console
$ uvx --from markdown markdown_py -x admonition test.md
<h1>List item with MkDocs admonition</h1>
<ul>
<li>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
  incididunt ut labore et dolore magna aliqua.</li>
</ul>
<p>!!! note</p>
<pre><code>    Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
</code></pre>
```

The manually reformatted content (see "Expected behavior") is correct MkDocs-flavored Markdown:

```console
$ uvx --from markdown markdown_py -x admonition test.md
<h1>List item with MkDocs admonition</h1>
<ul>
<li>
<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
  incididunt ut labore et dolore magna aliqua.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi
ut aliquip ex ea commodo consequat.</p>
</div>
</li>
</ul>
```

And `rumdl check` (without `--fix`) succeeds:

```console
uvx rumdl check test.md

Success: No issues found in 1 file (16ms)
```

So linting works fine, only `--fix` is causing trouble.

### Configuration

```toml
[global]
flavor = "mkdocs"
line-length = 80

[MD013]
reflow = true
```

### Operating system

Linux

**Comment by rvben** [comment]
Thanks for reporting this, @sisp. This was fixed in v0.1.30 (commit 72b7b4d9). Both problems you identified are resolved:

- The admonition header retains its original indentation
- The admonition body is now properly line-wrapped

Upgrading to v0.1.30 or later should resolve this for you.

**Comment by sisp** [comment]
Thank you very much for the super quick fix, @rvben! :bow:
